draft-irtf-cfrg-opaque.md

```diff
@@ -453,6 +453,17 @@ struct {
 } CleartextCredentials;
 ~~~
 
+The secret and cleartext credentials, along with the mode, are wrapped up in a
+`Credentials` structure, defined as follows:
+
+~~~
+struct {
+  InnerEnvelopeMode mode;
+  SecretCredentials secret_credentials;
+  CleartextCredentials cleartext_credentials;
+} Credentials;
```

Comment:
I believe this will be the only instance of a struct that we define in this document which never gets serialized. Why not just replace all instances with a tuple of (mode, secret_credentials, cleartext_credentials)? And note that the RecoverCredentials function doesn't actually need to return the cleartext_credentials or the mode...

Reply:
You could, yeah. This is just a helper type to wrap up everything in that tuple.